hack/charts/cluster-api-operator/templates/core-conditions.yaml

```diff
@@ -27,5 +27,7 @@ spec:
     {{- end }}
 {{- end }}
 {{- end }}
+{{- if hasKey (default (dict) $.Values.deploymentOverride ) "coreCondition" }}
```

Comment:
Can you add a test case for this manifest? Also, isn’t this a duplicate of `templates/core.yaml`

Reply:
what do you mean by duplicate? templates/core.yaml has the following:
https://github.com/kubernetes-sigs/cluster-api-operator/pull/525/files#diff-c73589d6ec69abbc84b0231a82d1fa536f3955e38026e4440db0b8e351295621R63

and I think we should avoid this default by just adding a default values to the values.yaml file.